src/Analyser/TypeSpecifier.php

```diff
@@ -133,6 +136,23 @@ public function specifyTypesInCondition(
 				return $this->create($exprNode, $type, $context);
 			}
 
+			$classType = $scope->getType($expr->class);
+			$type = TypeTraverser::map($classType, static function (Type $type, callable $traverse): Type {
+				if ($type instanceof UnionType || $type instanceof IntersectionType) {
+					return $traverse($type);
+				}
+				if ($type instanceof GenericClassStringType) {
+					return $type->getGenericType();
+				}
+				if ($type instanceof ConstantStringType) {
+					return new ObjectType($type->getValue());
+				}
+				return new ErrorType();
```

Comment:
Looks like this piece of code does not account for `$expr->class` to be of an object type. It also works in PHP: https://3v4l.org/SprYR